sdk/storage/azure_storage_blob/src/models/storage_error.rs

```diff
@@ -120,7 +119,7 @@ impl StorageError {
             additional_fields: HashMap<String, Value>,
         }
 
-        let xml_fields = azure_core::xml::read_xml::<StorageErrorXml>(xml_body)?;
+        let xml_fields = xml::from_xml::<_, StorageErrorXml>(xml_body)?;
```

Comment:
Why are you passing the XML body and the `raw_response`? That's all available via `raw_response.body().xml()`. Also, if you're just getting data out of the raw body to format, borrow strings for memory efficiency: https://serde.rs/lifetimes.html#borrowing-data-in-a-derived-impl. We do this for Azure standard errors. Don't waste memory.